hardware/raspberrypi/bcm2711_bootloader_config.md

```diff
@@ -84,28 +84,88 @@ Default: 0x00000001 (with 3 SD boot retries to match the current bootloader beha
 
 ### NET_BOOT_MAX_RETRIES
 Specify the maximum number of times that the bootloader will retry network boot.  
 -1 means infinite retries  
 Default: 0  
+Version: stable/pieeprom-2020-01-17.bin  
 
 ### DHCP_TIMEOUT
 The timeout in milliseconds for the entire DHCP sequence before failing the current iteration.  
 Default: 45000  
 Minimum: 5000  
+Version: stable/pieeprom-2020-01-17.bin  
 
 ### DHCP_REQ_TIMEOUT
 The timeout in milliseconds before retrying DHCP DISCOVER or DHCP REQ.  
 Default: 4000  
 Minimum: 500  
+Version: stable/pieeprom-2020-01-17.bin  
 
 ### TFTP_TIMEOUT
 The timeout in milliseconds for an individual file download via TFTP.  
 Default: 15000  
 Minimum: 5000  
+Version: stable/pieeprom-2020-01-17.bin  
 
 ### TFTP_IP
 Optional dotted decimal ip address (e.g. 192.169.1.99) for the TFTP server which overrides the server-ip from the DHCP request.  
 This maybe useful on home networks because tftpd-hpa can be used instead of dnsmasq where broadband router is the DHCP server.
 Default: ""  
+Version: stable/pieeprom-2020-01-17.bin  
+
+### TFTP_PREFIX
+In order to support unique TFTP boot directories for each Pi the bootloader prefixes the filenames with a device specific directory. If neither start4.elf nor start.elf are found in the prefixed directory then the prefix is cleared.
+On earlier models the serial number is used as the prefix, however, on Pi4 the MAC address is no longer generated from the serial number making it difficult to automatically create tftpboot directories on the server by inspecting DHCPDISCOVER packets. To support this the TFTP_PREFIX may be customized to either be the MAC address or a fixed value.
+
+* 0 - Use the serial number e.g. "9ffefdef/"
+* 1 - Use the string specified by TFTP_PREFIX_STR
+* 2 - Use the MAC address e.g. "DC-A6-32-01-36-C2/"
+Default: 0
+Version: stable/pieeprom-2020-01-17.bin  
+
+### TFTP_PREFIX_STR
+Specify the custom directory prefix string used when TFTP_PREFIX is set to 1. For example:- TFTP_PREFIX_STR=tftp_test/
+Default: ""
+Version: stable/pieeprom-2020-01-17.bin  
+
+## Network Boot
+### Server configuration                                                    
+Network boot requires a TFTP and NFS server to be configured.  See [Network boot server tutorial](bootmodes/net_tutorial.md)
+
+Additional notes:-
+* The MAC address on the Pi4 is programmed at manufacture and is not derived from the serial number.
+```                                                                          
+# mac adddress (ip addr) - it should start with DC:A6:32
+ip addr | grep ether | head -n1 | awk '{print $2}' | tr [a-z] [A-Z]                                                                                                                                                
+# serial number                                                                                                                         
+vcgencmd otp_dump | grep 28: | sed s/.*://g
```

Comment:
Perhaps the comments here should be changed to `# display mac address` and `# display serial number` ?
Also, you've got 3 'd's in address :wink:

Reply:
done